Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py

```diff
@@ -94,36 +94,43 @@ def fetch_by_ids_in_aql_query(self, aql_query: str, order_by: str = 'time'):
+    def fetch_by_aql_query(self, aql_query: str, max_fetch: int, after: datetime,
+                           order_by: str = 'time', from_param: None | int = None):
         """ Fetches events using AQL query.
 
         Args:
             aql_query (str): AQL query request parameter for the API call.
             max_fetch (int): Max number of events to fetch.
-            after (None | datetime): The date and time to fetch events from. Defaults to None.
+            after (None): The date and time to fetch events from.
             order_by (str): Order by parameter for the API call. Defaults to 'time'.
+            from_param (None | int): The next incident to start the fetch from. Defaults to None.
         Returns:
-            list[dict]: List of events objects represented as dictionaries.
+            (list[dict], int): A tuple with the List of events objects represented as dictionaries and the next event pointer.
         """
+        aql_query = f'{aql_query} after:{after.strftime(DATE_FORMAT)}'
         params: dict[str, Any] = {'aql': aql_query, 'includeTotal': 'true', 'length': max_fetch, 'orderBy': order_by}
-        if not after:  # this should only happen when get-events command is used without from_date argument
-            after = datetime.now() - timedelta(minutes=1)
-        params['aql'] += f' after:{after.strftime(DATE_FORMAT)}'  # add 'after' date filter to AQL query in the desired format
+        if from_param:
+            params['from'] = from_param
         raw_response = self.perform_fetch(params)
         results = raw_response.get('data', {}).get('results', [])
-
+        next = raw_response.get('data', {}).get('next') or 0
         # perform pagination if needed (until max_fetch limit),  cycle through all pages and add results to results list.
         # The response's 'next' attribute carries the index to start the next request in the
         # pagination (using the 'from' request parameter), or null if there are no more pages left.
-        while (next := raw_response.get('data', '').get('next')) and (len(results) < max_fetch):
-            if next < max_fetch:
-                params['length'] = max_fetch - next
-            params['from'] = next
-            raw_response = self.perform_fetch(params)
-            results.extend(raw_response.get('data', {}).get('results', []))
+        try:
+            while next and (len(results) < max_fetch):
+                if len(results) < max_fetch:
+                    params['length'] = max_fetch - len(results)
+                params['from'] = next
+                raw_response = self.perform_fetch(params)
+                next = raw_response.get('data', {}).get('next') or 0
+                current_results = raw_response.get('data', {}).get('results', [])
+                results.extend(current_results)
+                demisto.info(f"info-log: fetched {len(current_results)} results, total is {len(results)}, and {next=}.")
```

Comment:
maybe log here what type of log are we trying to retrieve, would be helpful

Reply:
I don't have a way to retrieve it from inside that function.
However, before calling this function, we log which event type we're attempting to retrieve so I think it's sufficient.